src/ex3/scripts/ItemManager.mjs

```diff
@@ -0,0 +1,202 @@
+    const after_task_deletion = this.tasks.length;
+    // task id is not in task length
+    if (current_task_length === after_task_deletion) { throw new Error('Task id illegal.'); }
+    this.file_manager.WriteToFileTasksArray(this.tasks);
+  }
+
+  /**
+     * send id to pokemon client and insert it to tasks array
+     * @param {int} id
+     */
+  async GetOnePokemon(id) {
+    // send the id to pokemon client
+    const result = Promise.resolve(this.pokemon_client.GetPokemonById(id));
+    // wait for response
+    await result;
+    return result;
+  }
+
+  /**
+     * parse string from input send it to pokemon client and insert it to tasks array
+     * @param {string} ids string from input
+     */
+  async GetMultiplePokemons(ids) {
+    const pokemon_ids = ids.split(',');
+    const result = Promise.resolve(this.pokemon_client.GetPokemonsByList(pokemon_ids));
+    await result;
+    return result;
+  }
+
+  /**
+     * inserts result to
+     * @param {string} result
+     * @param {int} id
+     */
+  InsertResultToArray(result) {
+    // check if found pokemon
+    if (typeof result.name !== 'undefined' && result.name !== 'Error') {
+      if (!this.CheckIfPokemonExists(result.id)) {
+        this.tasks.push({
+          name: result.name, images: result.images, id: result.id, completed: false,
+        });
+      }// found pokemon add the name and id of pokemon
+    } else if (typeof result === 'object') {
+      this.tasks.push({ data: result.message, id: 'Error', completed: false });
+    } else { this.tasks.push({ name: result, id: 'Regular task', completed: false }); }
+  }
+
+  /**
+     * clear all tasks
+     */
+  ClearArray() {
+    this.tasks = [];
+    this.file_manager.WriteToFileTasksArray(this.tasks);
+  }
+
+  /**
+     * sort tasks array
+     */
+  SortArrayByName() {
+    this.tasks.sort((a, b) => {
+      // tasks object has name or data for error
+      const item1 = a.name || a.data;
+      const item2 = b.name || b.data;
+      if (typeof item1 === 'undefined' || typeof item2 === 'undefined') { throw new Error('Task is undefined'); }
+      return item1.toLowerCase().localeCompare(item2.toLowerCase());
+    });
+    this.file_manager.WriteToFileTasksArray(this.tasks);
+    return true;
+  }
+
+  /**
+     * check if pokemon is in tasks
+     * @param {int} id id of pokemon to find
+     * @returns true if found false if not found
+     */
+  CheckIfPokemonExists(id) {
+    return this.tasks.find((task) => task.id === id);
+  }
+
+  /**
+     * parses response into pokemons array
+     * @param {Array} response
+     */
+  ParsePokemonNamesResponse(response) {
+    if (response.name !== 'Error') {
+      response.forEach((pokemon) => {
+      // split url string and get rid of empty elements
+        const parsed_url = pokemon.url.split('/').filter((item) => item);
+        const id = parsed_url[parsed_url.length - 1];
+        this.pokemons.push({ name: pokemon.name, id });
```

Comment:
Couldn't you get the name from the response itself and not from the URL?

Reply:
the response to this API call is an array of objects 
{
"name": STRING,
"url": STERING
}
so I get the name from the object and the id by parsing the url filed